server/encoder/video_encoder_vulkan.cpp

```diff
@@ -745,8 +745,12 @@ std::pair<bool, vk::Semaphore> wivrn::video_encoder_vulkan::present_image(vk::Im
 		                .flags = vk::VideoCodingControlFlagBitsKHR::eEncodeRateControl,
 		        });
 
+	// avoid self-reference when the DPB ring wraps (cannot reference the slot we're writing).
+	auto curr_slot_idx = slot->info.slotIndex;
+	auto use_ref = (ref_slot && ref_slot->info.slotIndex == curr_slot_idx);
+
```

Comment:
this part avoids a case where the encoder will get stuck not producing any frames until an idr is requested.

Reply:
possible fix for https://github.com/WiVRn/WiVRn/issues/303 but in my case i was just shown the "launch an app" panel, no artifacts.